internal/config/config_test.go

```diff
@@ -0,0 +1,175 @@
+// +build envtest
+
+/*
+ Copyright 2021 Crunchy Data Solutions, Inc.
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package config
+
+import (
+	"context"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/crunchydata/postgres-operator/internal/controller/runtime"
+	"github.com/crunchydata/postgres-operator/internal/naming"
+	"github.com/crunchydata/postgres-operator/pkg/apis/postgres-operator.crunchydata.com/v1beta1"
+	"gotest.tools/v3/assert"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/envtest"
+)
+
+func TestSetContainerImage(t *testing.T) {
+
+	t.Run("env var not set and spec empty", func(t *testing.T) {
+
+		assert.Equal(t, SetContainerImage("", CrunchyPostgresHA_PG10), "")
+	})
+
+	t.Run("image name from spec", func(t *testing.T) {
+
+		os.Setenv(CrunchyPostgresHA_PG10, "envVarImage")
+		assert.Equal(t, SetContainerImage("specImageName", CrunchyPostgresHA_PG10), "specImageName")
+	})
+
+	t.Run("get image from env var when image missing in spec", func(t *testing.T) {
+
+		os.Setenv(CrunchyPostgresHA_PG10, "envVarImage")
+		assert.Equal(t, SetContainerImage("", CrunchyPostgresHA_PG10), "envVarImage")
+	})
+
+	t.Run("env var and spec empty", func(t *testing.T) {
+
+		os.Setenv(CrunchyPostgresHA_PG10, "")
+		assert.Equal(t, SetContainerImage("", CrunchyPostgresHA_PG10), "")
+	})
+}
+
+func TestSetPGContainerImage(t *testing.T) {
+
+	// setup the test environment and ensure a clean teardown
+	tEnv, tClient, _ := setupTestEnv(t, "postgrescluster-controller")
+	ctx := context.Background()
+	// set namespace name
+	ns := &v1.Namespace{}
+	ns.GenerateName = "postgres-operator-test-"
+	assert.NilError(t, tClient.Create(ctx, ns))
+	t.Cleanup(func() {
+		assert.Check(t, tClient.Delete(ctx, ns))
+		teardownTestEnv(t, tEnv)
+	})
+	namespace := ns.Name
+
+	// set up cluster1
+	clusterName1 := "hippocluster1"
+
+	// set up test cluster1
+	cluster1 := &v1beta1.PostgresCluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      clusterName1,
+			Namespace: namespace,
+		},
+		Spec: v1beta1.PostgresClusterSpec{
+			Metadata:        &v1beta1.Metadata{},
+			PostgresVersion: 12,
+			InstanceSets:    []v1beta1.PostgresInstanceSetSpec{},
+		},
+	}
+
+	cluster1.SetGroupVersionKind(v1.SchemeGroupVersion.WithKind("postgrescluster"))
+	if err := tClient.Create(ctx, cluster1); err != nil {
```

Comment:
♻️ I suspect we don't need an entire Kubernetes for these tests because the code does not call the API. If so, drop the `+build envtest`, too.

Reply:
Yes, that seems to be the case with these tests, I'll remove the envtest portions.